packages/react-core/src/components/Form/FormGroup.tsx

```diff
@@ -60,23 +63,30 @@ export const FormGroup: React.FunctionComponent<FormGroupProps> = ({
   );
 
   return (
-    <div {...props} className={css(styles.formGroup, isInline ? styles.modifiers.inline : className)}>
+    <div {...props} className={css(styles.formGroup, className)}>
       {label && (
-        <label className={css(styles.formLabel, hasNoPaddingTop && styles.modifiers.noPaddingTop)} htmlFor={fieldId}>
-          <span className={css(styles.formLabelText)}>{label}</span>
-          {isRequired && (
-            <span className={css(styles.formLabelRequired)} aria-hidden="true">
-              {ASTERISK}
-            </span>
-          )}
-        </label>
+        <div className={css(styles.formGroupLabel, hasNoPaddingTop && styles.modifiers.noPaddingTop)}>
+          <label className={css(styles.formLabel)} htmlFor={fieldId}>
+            <span className={css(styles.formLabelText)}>{label}</span>
+            {isRequired && (
+              <span className={css(styles.formLabelRequired)} aria-hidden="true">
+                {' '}
+                {ASTERISK}
+              </span>
+            )}
+          </label>{' '}
+          {React.isValidElement(labelIcon) && labelIcon}
+        </div>
       )}
-      {children}
-      {validated === ValidatedOptions.error && helperTextInvalid
-        ? inValidHelperText
-        : validated !== ValidatedOptions.error && helperText
-        ? validHelperText
-        : ''}
+
+      <div className={css(styles.formGroupControl, isInline && styles.modifiers.inline)}>
+        {children}
+        {validated === ValidatedOptions.error && helperTextInvalid
```

Comment:
I'm not a fan of nested ternaries. Do you think it'd be possible to clean this up into a `helperText` variable?